File: src/102.binary-tree-level-order-traversal.py
#
# @lc app=leetcode id=102 lang=python3
#
# [102] Binary Tree Level Order Traversal
#
from collections import deque
from typing import List
# @lc code=start
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []

        answer = []
        stack = deque([root])
        while len(stack) > 0:
            temp = []
            print(stack)
            for i in range(len(stack)):
                current = stack.popleft()

                temp.append(current.val)
                
                if current.left:
                    stack.append(current.left)
                if current.right:
                    stack.append(current.right)
            answer.append(temp)
        
        return answer

# @lc code=end

File: Binary Tree/src/102.binary-tree-level-order-traversal.js
/*
 * @lc app=leetcode id=102 lang=javascript
 *
 * [102] Binary Tree Level Order Traversal
 */

// Definition for a binary tree node.
class TreeNode {
  consturctor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}
// @lc code=start
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
  if (root === null) {
    return [];
  }

  const history = [];
  const stack = [root];
  let level = [];

  let node;
  let stackLength;
  while (stack.length > 0) {
    stackLength = stack.length;
    level = [];
    for (var i = 0; i < stackLength; i++) {
      node = stack.pop();
      level.push(node.val);

      if (node.left) {
        stack.unshift(node.left);
      }
      if (node.right) {
        stack.unshift(node.right);
      }
    }

    history.push(level);
  }
   
  return history;
};
// @lc code=end

